Leverage TensorPipe's automatic SHM address selection

lw opened the pull request:
Summary: TensorPipe until now required PyTorch to come up and provide a unique identifier to use as address for the UNIX domain socket used in the SHM transport. However the Linux kernel can automatically assign an available address (like it does with IP ports), and TensorPipe now supports it, so we can remove that useless PyTorch logic.

Test Plan: CI

Differential Revision: D30220352



cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd @cbalioglu @gcramer23

mrshenli reviewed (changes requested):
oops, test failures are real:

```
RuntimeError: In addrImplFromLoop at tensorpipe/transport/shm/listener_impl.cc:97 "errorgetsockname: Bad file descriptor (this error originated at tensorpipe/common/socket.cc:129)"

```

lw committed:
Update TensorPipe submodule

Differential Revision: D30573500

fbshipit-source-id: b0e400d6266ab0a282e13f0cd387198e8d908bc8

lw committed:
Leverage TensorPipe's automatic SHM address selection (#63028)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/63028

TensorPipe until now required PyTorch to come up and provide a unique identifier to use as address for the UNIX domain socket used in the SHM transport. However the Linux kernel can automatically assign an available address (like it does with IP ports), and TensorPipe now supports it, so we can remove that useless PyTorch logic.

Test Plan: CI

Differential Revision: D30220352

fbshipit-source-id: 07642ebc9eb30386ed875530ceda4abdecf3cd23

mrshenli reviewed (approved):
LGTM!
